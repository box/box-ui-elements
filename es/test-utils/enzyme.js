function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({
  children
}) => {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: defaultTheme
  }, children);
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
//# sourceMappingURL=enzyme.js.map