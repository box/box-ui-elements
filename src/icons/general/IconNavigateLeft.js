// @flow
import * as React from 'react';

import AccessibleSVG from '../accessible-svg';

type Props = {
    className?: string,
    height?: number,
    /** A text-only string describing the icon if it's not purely decorative for accessibility */
    title?: string | React.Element<any>,
    width?: number,
};

const IconNavigateLeft = ({ className = '', height = 48, title, width = 48 }: Props) => (
    <AccessibleSVG
        className={`icon-navigate-left ${className}`}
        focusable="false"
        title={title}
        width={width}
        height={height}
        viewBox="0 0 48 48"
    >
        <path
            fill="#494949"
            stroke="#DCDCDC"
            strokeMiterlimit="10"
            d="M30.8,33.2L21.7,24l9.2-9.2L28,12L16,24l12,12L30.8,33.2z"
        />
        <path display="none" fill="none" d="M0,0h48v48H0V0z" />
    </AccessibleSVG>
);

export default IconNavigateLeft;
