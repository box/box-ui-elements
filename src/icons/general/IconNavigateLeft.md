Large chevron pointing left with a light border but without focus and not themable

```js
<IconNavigateLeft />
```
