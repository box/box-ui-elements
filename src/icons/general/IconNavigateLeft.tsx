import * as React from 'react';

import AccessibleSVG from '../accessible-svg';

import { TwoTonedIcon } from '../iconTypes';

const IconNavigateLeft = ({ className = '', height = 48, title, width = 48 }: TwoTonedIcon) => (
    <AccessibleSVG
        className={`icon-navigate-left ${className}`}
        focusable="false"
        height={height}
        title={title}
        viewBox="0 0 48 48"
        width={width}
    >
        <path
            d="M30.8,33.2L21.7,24l9.2-9.2L28,12L16,24l12,12L30.8,33.2z"
            fill="#494949"
            stroke="#DCDCDC"
            strokeMiterlimit="10"
        />
        <path d="M0,0h48v48H0V0z" display="none" fill="none" />
    </AccessibleSVG>
);

export default IconNavigateLeft;
