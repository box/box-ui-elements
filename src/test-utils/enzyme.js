/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, { wrappingComponent: Wrappers, ...options });

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, { wrappingComponent: Wrappers, ...options });

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
