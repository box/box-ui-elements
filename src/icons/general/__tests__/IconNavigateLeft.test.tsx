import React from 'react';
import { shallow } from 'enzyme';

import IconNavigateLeft from '../IconNavigateLeft';

describe('icons/general/IconNavigateLeft', () => {
    const getWrapper = (props = {}) => shallow(<IconNavigateLeft {...props} />);

    test('should correctly render default icon', () => {
        const wrapper = getWrapper();

        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render icon with specified props', () => {
        const wrapper = getWrapper({
            className: 'test',
            height: 100,
            title: 'title',
            width: 200,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
