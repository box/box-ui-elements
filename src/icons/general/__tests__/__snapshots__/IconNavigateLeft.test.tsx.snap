// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons/general/IconNavigateLeft should correctly render default icon 1`] = `
<AccessibleSVG
  className="icon-navigate-left "
  focusable="false"
  height={48}
  viewBox="0 0 48 48"
  width={48}
>
  <path
    d="M30.8,33.2L21.7,24l9.2-9.2L28,12L16,24l12,12L30.8,33.2z"
    fill="#494949"
    stroke="#DCDCDC"
    strokeMiterlimit="10"
  />
  <path
    d="M0,0h48v48H0V0z"
    display="none"
    fill="none"
  />
</AccessibleSVG>
`;

exports[`icons/general/IconNavigateLeft should correctly render icon with specified props 1`] = `
<AccessibleSVG
  className="icon-navigate-left test"
  focusable="false"
  height={100}
  title="title"
  viewBox="0 0 48 48"
  width={200}
>
  <path
    d="M30.8,33.2L21.7,24l9.2-9.2L28,12L16,24l12,12L30.8,33.2z"
    fill="#494949"
    stroke="#DCDCDC"
    strokeMiterlimit="10"
  />
  <path
    d="M0,0h48v48H0V0z"
    display="none"
    fill="none"
  />
</AccessibleSVG>
`;
